module/VuFind/src/VuFind/Db/Row/User.php

```diff
@@ -275,28 +275,34 @@ public function getTags($resourceId = null, $listId = null, $source = null)
     }
 
     /**
-     * Same as getTags(), but returns a string for use in edit mode rather than an
-     * array of tag objects.
+     * Get tags assigned by the user to a favorite list.
      *
-     * @param string $resourceId Filter for tags tied to a specific resource (null
-     * for no filter).
-     * @param int    $listId     Filter for tags tied to a specific list (null for no
-     * filter).
-     * @param string $source     Filter for tags tied to a specific record source
-     * (null for no filter).
+     * @param int $listId List id
+     *
+     * @return \Laminas\Db\ResultSet\AbstractResultSet
+     */
+    public function getListTags($listId)
+    {
+        return $this->getDbTable('Tags')
+            ->getForList($listId, $this->id);
+    }
+
+    /**
+     * Returns a string for use in edit mode.
+     *
+     * @param array $tags Tags (output from getTags)
      *
      * @return string
      */
-    public function getTagString($resourceId = null, $listId = null, $source = null)
+    public function getTagString($tags)
```

Comment:
Rather than modifying getTagString(), maybe it would be better to refactor this, so there's a new formatTagString method that gets used by getTagString and can also be called separately. That would introduce fewer backward compatibility breaks. We can always deprecate getTagString() for future removal if it's no longer useful.